### DISCUSSION TITLE
Does frigate update the object types over the course of an event?

### DISCUSSION THREAD

**Discussion by nuansed** [blakeblackshear/frigate#3391] [Ask A Question] [answered]
Say a person enters the frame from far away, and is first incorrectly detected as a cat. As the person comes closer, the person is finally detected accurately. How does frigate handle this change? Does frigate update the Event ID, or does it just end the previous event, then create a new event with every object change?

**Comment by NickM-27** [comment, marked as answer]
The objects are detected separately and keep separate but same IDs (as in not changing) throughout their own detection. 

That being said, it takes multiple detection with a median score to be an event (versus just a detection) so it won't be swapping back and forth.

Here's a great part of the docs to read that explains the way an object is confirmed based on multiple frames https://docs.frigate.video/guides/false_positives/
